crates/pop-cli/src/commands/build/spec.rs

```diff
@@ -0,0 +1,381 @@
+		if is_supported(None)? {
+			// If para id has been provided we can build the spec
+			// otherwise, we need to guide the user.
+			let _ = match self.id {
+				Some(_) => self.build(&mut cli::Cli),
+				None => {
+					let config = guide_user_to_generate_spec().await?;
+					config.build(&mut cli::Cli)
+				},
+			};
+			return Ok("spec");
+		} else {
+			cli::Cli.outro_cancel(
+				"🚫 Can't build a specification for target. Maybe not a chain project ?",
+			)?;
+			Ok("spec")
+		}
+	}
+
+	/// Builds a parachain spec.
+	///
+	/// # Arguments
+	/// * `cli` - The CLI implementation to be used.
+	fn build(self, cli: &mut impl cli::traits::Cli) -> anyhow::Result<&'static str> {
+		cli.intro(format!("Building your chain spec"))?;
+
+		// Either a para id was already provided or user has been guided to provide one.
+		let para_id = self.id.unwrap_or(2000);
+		// Notify user in case we need to build the parachain project.
+		if !self.release {
+			cli.warning("NOTE: this command defaults to DEBUG builds for development chain types. Please use `--release` (or simply `-r` for a release build...")?;
+			#[cfg(not(test))]
+			sleep(Duration::from_secs(3))
+		}
+
+		let spinner = cliclack::spinner();
+		spinner.start("Generating chain specification...");
+
+		// Create output path if needed
+		let mut output_path = self.output_file.unwrap_or_else(|| PathBuf::from("./"));
+		let mut plain_chain_spec;
+		if output_path.is_dir() {
+			if !output_path.exists() {
+				// Generate the output path if needed
+				create_dir_all(&output_path)?;
+			}
+			plain_chain_spec = output_path.join(DEFAULT_SPEC_NAME);
+		} else {
+			plain_chain_spec = output_path.clone();
+			output_path.pop();
+			if !output_path.exists() {
+				// Generate the output path if needed
+				create_dir_all(&output_path)?;
+			}
+		}
+		plain_chain_spec.set_extension("json");
+
+		// Locate binary, if it doesn't exist trigger build.
+		let mode: Profile = self.release.into();
+		let cwd = current_dir().unwrap_or(PathBuf::from("./"));
+		let binary_path = match binary_path(&mode.target_folder(&cwd), &cwd.join("node")) {
+			Ok(binary_path) => binary_path,
+			_ => {
+				cli.info(format!("No node was not found. The project will be built locally."))?;
+				cli.warning("NOTE: this may take some time...")?;
+				build_parachain(&cwd, None, &mode, None)?
+			},
+		};
+
+		// Generate plain spec.
+		spinner.set_message("Generating plain chain specification...");
+		let mut generated_files =
+			vec![format!("Specification and artifacts generated at: {}", &output_path.display())];
+		generate_plain_chain_spec(&binary_path, &plain_chain_spec, para_id, self.default_bootnode)?;
+		generated_files.push(format!(
+			"Plain text chain specification file generated at: {}",
+			plain_chain_spec.display()
+		));
+
+		// Customize spec based on input.
+		let relay = self.relay.unwrap_or(RelayChain::PaseoLocal).to_string();
+		replace_relay_spec(&plain_chain_spec, &relay, "rococo-local")?;
```

Comment:
There are no guarantees that it will always be `rococo-local` to replace. I maintain that this should be done properly via serde or some json editing crate.

Reply:
This can be resolved too no?